lib/flipper/feature.rb

```diff
@@ -100,9 +100,14 @@ def clear
     #
     # Returns true if enabled, false if not.
     def enabled?(*actors)
-      # Allows null object pattern. See PR for more.
-      # https://github.com/flippercloud/flipper/pull/887
-      actors = actors.flatten.reject(&:nil?).map { |actor| Types::Actor.wrap(actor) }
+      actors = Array(actors).
+        # Avoids to_ary warning that happens when passing DelegateClass of an 
+        # ActiveRecord object and using flatten here. This is tested in 
+        # spec/flipper/model/active_record_spec.rb.
+        flat_map { |actor| actor.is_a?(Array) ? actor : [actor] }. 
```

Comment:
I like the `flat_map` trick to avoid the `to_ary` call of `flatten`. The only downside is that it now doesn't handle super deeply nested arrays, but I guess this is less of a problem? I suppose this is mostly to handle passing the list of actors as a simple array. 

That being said, you can simplify this a bit and avoid unnecessary array allocations.

```suggestion
    # Avoids to_ary warning that happens when passing DelegateClass of an 
    # ActiveRecord object and using flatten here. This is tested in 
    # spec/flipper/model/active_record_spec.rb.
    actors.flat_map(&:itself)
```

`actors` is already an array, because of `*actors` in the method definition. `flat_map` doesn't need an array as a return value, it'll deal with regular objects just fine. 

`itself` is available since Ruby 2.2.0, so if you have to support older versions it should probably be a `flat_map { |actor| actor }`

Reply:
> The only downside is that it now doesn't handle super deeply nested arrays, but I guess this is less of a problem?

Yeah, deeply nested should never happen. I'd consider that an argument error. 

> That being said, you can simplify this a bit and avoid unnecessary array allocations.

Sweet! I'll try that.

Reply:
Really? 🤔 I remember giving this a try and "it worked on my machine" 🙈